meteor/client/lib/forms/schemaFormTable.tsx

```diff
@@ -0,0 +1,325 @@
+import { faCheck, faPencilAlt, faPlus, faTrash } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { clone, literal, objectPathGet, objectPathSet } from '@sofie-automation/corelib/dist/lib'
+import classNames from 'classnames'
+import React, { useCallback, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import { WrappedOverridableItemNormal, OverrideOpHelper } from '../../ui/Settings/util/OverrideOpHelper'
+import { useToggleExpandHelper } from '../../ui/Settings/util/ToggleExpandedHelper'
+import { doModalDialog } from '../ModalDialog'
+import { getSchemaSummaryFieldsForObject, SchemaSummaryField, translateStringIfHasNamespaces } from './schemaFormUtil'
+import { SchemaFormWithOverrides } from './schemaFormWithOverrides'
+import { JSONSchema } from '@sofie-automation/shared-lib/dist/lib/JSONSchemaTypes'
+import { getSchemaDefaultValues } from '@sofie-automation/shared-lib/dist/lib/JSONSchemaUtil'
+
+interface SchemaFormTableProps {
+	schema: JSONSchema
+	translationNamespaces: string[]
+
+	attr: string
+
+	item: WrappedOverridableItemNormal<any>
+	overrideHelper: OverrideOpHelper
+}
+export const SchemaFormTable = ({
+	schema,
+	translationNamespaces,
+	attr,
+	item,
+	overrideHelper,
+}: SchemaFormTableProps) => {
+	const rows = useMemo(() => (attr ? objectPathGet(item.computed, attr) : item.computed) || [], [attr, item.computed])
+
+	const addNewItem = useCallback(() => {
+		// Build the new object
+		const newObj = clone(rows)
+		newObj.push(getSchemaDefaultValues(schema.items))
+
+		// Send it onwards
+		overrideHelper.setItemValue(item.id, attr, newObj)
+	}, [schema.items, overrideHelper, rows, item.id])
+
+	const tableOverrideHelper = useMemo(
+		() => new OverrideOpHelperTable(overrideHelper, item.id, rows, attr),
+		[overrideHelper, item.id, rows, attr]
+	)
+
+	const title = schema['ui:title']
```

Comment:
I think it would be good to define some sort of an `enum` or object with these special extensions to JSONSchema.